app/src/main/java/com/amaze/filemanager/utils/MainActivityHelper.java

```diff
@@ -404,7 +405,10 @@ public void invalidName(final HybridFile file) {
   // For Android 5
   public static final int CAN_CREATE_FILES = 2;
 
-  public int checkFolder(final File folder, Context context) {
+  @IntDef({DOESNT_EXIST, WRITABLE_OR_ON_SDCARD, CAN_CREATE_FILES})
+  public @interface FolderState {}
```

Comment:
Suggested to declare these constants inside the interface, and maybe a separate kotlin class instead of nested?

Reply:
> declare these constants inside the interface

I'm not sure that's legal.

Reply:
In kotlin "members are not allowed in annotation class".